plugins/continuous-toolbox/src/ContinuousFlyout.js

```diff
@@ -185,4 +200,152 @@ export class ContinuousFlyout extends Blockly.VerticalFlyout {
+    const oldBlocks = this.recycleBlocks_;
+    this.recycleBlocks_ = [];
+    for (const oldBlock of oldBlocks) {
+      oldBlock.dispose(false, false);
+    }
+  }
+
+  /**
+   * @override
+   */
+  createBlock_(blockXml) {
+    const blockType = blockXml.getAttribute('type');
+    const blockIdx = this.recycleBlocks_.findIndex(function(block) {
+      return block.type === blockType;
+    });
+    let curBlock;
+    if (blockIdx > -1) {
+      curBlock = this.recycleBlocks_.splice(blockIdx, 1)[0];
+    } else {
+      curBlock = Blockly.Xml.domToBlock(blockXml, this.workspace_);
+    }
+    if (!curBlock.isEnabled()) {
+      // Record blocks that were initially disabled.
+      // Do not enable these blocks as a result of capacity filtering.
+      this.permanentlyDisabled_.push(curBlock);
+    }
+    return curBlock;
+  }
+
+  /**
+   * @override
+   */
+  clearOldBlocks_() {
+    // Delete any blocks from a previous showing.
+    const oldBlocks = /** @type {!Array<!Blockly.BlockSvg>} */
+        (this.workspace_.getTopBlocks(false));
+    for (const block of oldBlocks) {
+      if (block.workspace == this.workspace_) {
+        if (this.recyclingEnabled_ &&
+          this.blockIsRecyclable_(block)) {
+          this.recycleBlock_(block);
+        } else {
+          block.dispose(false, false);
+        }
+      }
+    }
+    // Delete any mats from a previous showing.
+    for (const rect of this.mats_) {
+      if (rect) {
+        Blockly.Tooltip.unbindMouseEvents(rect);
+        Blockly.utils.dom.removeNode(rect);
+      }
+    }
+    this.mats_.length = 0;
+    // Delete any buttons from a previous showing.
+    for (const button of this.buttons_) {
+      button.dispose();
+    }
+    this.buttons_.length = 0;
+
+    // Clear potential variables from the previous showing.
+    this.workspace_.getPotentialVariableMap().clear();
+  }
+
+  /**
+   * Determine if this block can be recycled in the flyout.  Blocks that have no
+   * variables and are not dynamic shadows can be recycled.
+   * @param {!Blockly.BlockSvg} block The block to attempt to recycle.
+   * @return {boolean} True if the block can be recycled.
+   * @protected
+   */
+  blockIsRecyclable_(block) {
+    // If the block needs to parse mutations, never recycle.
+    if (block.mutationToDom && block.domToMutation) {
+      return false;
+    }
+
+    for (const input of block.inputList) {
+      for (const field of input.fieldRow) {
+        // No variables.
+        if (field instanceof Blockly.FieldVariable) {
+          return false;
+        }
+        if (field instanceof Blockly.FieldDropdown) {
+          if (field.isOptionListDynamic()) {
+            return false;
+          }
+        }
+      }
+      // Check children.
+      if (input.connection) {
+        const targetBlock = /** @type {Blockly.BlockSvg} */
+            (input.connection.targetBlock());
+        if (targetBlock && !this.blockIsRecyclable_(targetBlock)) {
```

Comment:
Is `this` bound correctly at this point? Please check that calling `this.blockIsRecyclable_` here actually works as intended if you've also called `setBlockIsRecyclable` to change the function.

Reply:
Just tested re setting this method using `setBlockIsRecyclable` and the `this` looks like it is is bound correctly.